internal/services/netapp/netapp_volume_group_sap_hana_resource.go

```diff
@@ -0,0 +1,628 @@
+			replicationClient := metadata.Client.NetApp.VolumeReplicationClient
+
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var model NetAppVolumeGroupSapHanaModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			id := volumegroups.NewVolumeGroupID(subscriptionId, model.ResourceGroupName, model.AccountName, model.Name)
+
+			metadata.Logger.Infof("Import check for %s", id)
+			existing, err := client.VolumeGroupsGet(ctx, id)
+			if err != nil && existing.HttpResponse.StatusCode != http.StatusNotFound {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+
+			if existing.Model != nil && existing.Model.Id != nil && *existing.Model.Id != "" {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			applicationType := volumegroups.ApplicationTypeSAPNegativeHANA
+
+			volumeList, err := expandNetAppVolumeGroupVolumes(model.Volumes)
+			if err != nil {
+				return err
+			}
+
+			// Performing some basic validations that are not possible in the schema
+			if errorList := validateNetAppVolumeGroupVolumes(volumeList, applicationType); len(errorList) > 0 {
+				return fmt.Errorf("one or more issues found while performing deeper validations for %s:\n%+v", id, errorList)
+			}
+
+			// Parse volume list to set secondary volumes for CRR
+			for i, volumeCrr := range *volumeList {
+				if volumeCrr.Properties.DataProtection != nil &&
+					volumeCrr.Properties.DataProtection.Replication != nil &&
+					strings.EqualFold(string(*volumeCrr.Properties.DataProtection.Replication.EndpointType), string(volumegroups.EndpointTypeDst)) {
+
+					// Modify volumeType as data protection type on main volumeList
+					// so it gets created correctly as data protection volume
+					(*volumeList)[i].Properties.VolumeType = utils.String("DataProtection")
+				}
+			}
+
+			// TODO: deploymentSpecId is temporary until the backend is updated and deploymentSpecId is not required anymore,
+			//       it will be handled internally by the RP
+			deploymentSpecId := "20542149-bfca-5618-1879-9863dc6767f1"
+
+			parameters := volumegroups.VolumeGroupDetails{
+				Location: utils.String(location.Normalize(model.Location)),
+				Properties: &volumegroups.VolumeGroupProperties{
+					GroupMetaData: &volumegroups.VolumeGroupMetaData{
+						GroupDescription:      utils.String(model.GroupDescription),
+						ApplicationType:       &applicationType,
+						ApplicationIdentifier: utils.String(model.ApplicationIdentifier),
+						DeploymentSpecId:      utils.String(deploymentSpecId),
+					},
+					Volumes: volumeList,
+				},
+			}
+
+			err = client.VolumeGroupsCreateThenPoll(ctx, id, parameters)
+			if err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			// Waiting for volume group be completely provisioned
+			if err := waitForVolumeGroupCreateOrUpdate(ctx, client, id); err != nil {
```

Comment:
`VolumeGroupsCreateThenPoll` should do the job of waiting until the resource is fully created before moving on. Did you see something that made CreateThenPoll return earlier than it should? If so, can you open an issue on [pandora](https://github.com/hashicorp/pandora) so we can get that resolved.

Reply:
Did you somehow implemented the same functionality of the wait functions within the XYZCreateThenPoll? Which is actually performing GETS instead of only relying on status of the PUT operation?

If not, then this must still be there, because for most of our resources, if you don't perform GETs to check that the actual resource is there after the PUT request is done ARM will say that the resouce is not found if you need further operations on the resource that requires it to be fully available.

Reply:
@paulomarquesc not right now, but that's tracked in https://github.com/hashicorp/go-azure-sdk/issues/307

Reply:
Thanks @tombuildsstuff , this will be awesome to have and then we will be able to remove the wait functions.